src/qibolab/platform.py

```diff
@@ -30,88 +39,56 @@ def __init__(self, name, runcard, instruments, channels):
         log.info("Loading platform %s", name)
 
         self.name = name
-        self.runcard = runcard
+        self.is_connected = False
         self.instruments: List[Instrument] = instruments
         self.channels: ChannelMap = channels
 
-        self.qubits: Dict[QubitId, Qubit] = {}
-        self.pairs: Dict[Tuple[QubitId, QubitId], QubitPair] = {}
-
-        # Values for the following are set from the runcard in ``reload_settings``
-        self.settings = None
-        self.is_connected = False
-
-        self.nqubits = None
-        self.resonator_type = None
-        self.topology = None
-
-        self.nshots = None
-        self.relaxation_time = None
-        self.sampling_rate = None
-
-        # TODO: Remove this (needed for the multiqubit platform)
-        self.native_gates = {}
-        self.two_qubit_native_types = NativeType(0)
-        # Load platform settings
-        self.reload_settings()
-
-    def __repr__(self):
-        return self.name
-
-    def _check_connected(self):
-        if not self.is_connected:  # pragma: no cover
-            raise_error(RuntimeError, "Cannot access instrument because it is not connected.")
-
-    def reload_settings(self):
-        # TODO: Remove ``self.settings``
-        if self.settings is None:
-            # Load initial configuration
-            if isinstance(self.runcard, dict):
-                settings = self.settings = self.runcard
-            else:
-                with open(self.runcard) as file:
-                    settings = self.settings = yaml.safe_load(file)
+        # Load initial configuration
+        if isinstance(runcard, dict):
```

Comment:
If possible, accept only one type of `runcard`.

And I'd recommend a `dict`, since you're always on time to load it before (if you have a path), but it would be annoying to dump it, in case you already have a `dict` (if you don't need the disk and file system, you should be able to operate without).

Reply:
Actually the idea is to not accept a `runcard` at all. The user will have to create the `instruments` and `qubits` in the `create` method and return the platform as `Platform(instruments, qubits)`. This would require creating some helper functions in qibolab to assist with the creation of `Qubit` and `QubitPair` objects, which in this PR happens in `Platform.__init__`. `platform.dump` may also need to be disentangled from the platform and be another helper method, but this is optional.

The advantage of this is that `Platform` will be much simpler and not constrained by a specific serialization. Someone could even create a qubit and an instrument and use it directly in a single python script without any yaml or database. The disadvantage is that `create` methods may be more complex and with code repetition among them.

This is another big (conceptually, not in terms of code diff) change, so I will do it in another PR, after this.

Reply:
Should we open an issue for this?

Reply:
> Should we open an issue for this?

Done.